### PR TITLE
Fix for modern node & npm

### PR DESCRIPTION
**Pull request by AaronDewes**
This includes various fixes for modern npm (Which does not have the API from NPM 7 anymore), adds yarn support, moves create-probot-app to ESM, and requires Node 16.

Requiring Node 16 for create-probot-app should help encourage people to use a modern node for new apps, so migratting probot to require Node 14 or 16 in the future should be easier.

### CODE REVIEW

File: src/helpers/user-interaction.ts [hunk 1]
```diff
@@ -197,7 +201,7 @@ export async function runCliManager(): Promise<CliConfig> {
   // TSC mangles output directory when using normal import methods for
   // package.json. See
   // https://github.com/Microsoft/TypeScript/issues/24715#issuecomment-542490675
-  const pkg = require(require.resolve("../../package.json"));
+  const pkg = JSON.parse(fs.readFileSync(path.join(__dirname, "..", "..", "package.json"), "utf-8"));
```
Comment (wolfy1339) [inline src/helpers/user-interaction.ts:204]: You cannot use `__dirname` with ESM. It doesn't exist in that context

**Comment by devi-prsd** [comment]
+1

**Comment by AaronDewes** [comment]
@gr2m Any chance you can give this a review?

**Review by gr2m — APPROVED** [review]
Thank you!

**Comment by gr2m** [comment]
We can drop support for Node 16 as well btw, we did the same in @octokit. Node 16 reached end-of-life, it will no longer receive updates, not even security updates: https://nodejs.dev/en/about/releases/

**Pull request merged**